### PR TITLE
Player argument to /transferserver command

### LINKED ISSUE #4065: Transferserver should take optional [player] argument
## Description
An optional `[player]` argument for `/transferserver`

## Justification
I have seen BDS modding communities like ElementZero or BDSX doing that to allow transferring specific players to servers (i.e. via NPCs)

Here is EZ's implementation for reference: https://github.com/Element-0/ElementZero/blob/core/BuiltinMods/Essentials/transfer_command.cpp

## Alternative methods
/transferplayer command

### PR DESCRIPTION
**Pull request by ShockedPlot7560**
## Introduction

The /transferserver command could only be used by the person himself, and not transferred to any player.
### Relevant issues

Fixes: pmmp/PocketMine-MP#4065

### API changes

N/A
### Behavioural changes

N/A
## Backwards compatibility

N/A
## Follow-up

Requires translations:

| Name | Value in eng.ini |
| :--: | :---: |
| `pocketmine.command.transferserver.description` | `Transfer someone to another server` |
| `pocketmine.command.transferserver.usage` | `/transferserver <server> [port] [player]` |

## Tests
Connect a second account and transfer it to a server of your choice. Works

### CODE REVIEW

**Review by dktapps — CHANGES_REQUESTED** [review]
There are a number of problems with this PR:
- It allows any player to transfer any other player, regardless of permissions.
- It doesn't properly handle invalid players. If a player is specified who's offline, an error message will be sent, and then the server will crash anyway.
- The formatting doesn't follow PocketMine-MP's code style.

**Comment by ShockedPlot7560** [comment]
> There are a number of problems with this PR:
> 
> * It allows any player to transfer any other player, regardless of permissions.
> * It doesn't properly handle invalid players. If a player is specified who's offline, an error message will be sent, and then the server will crash anyway.
> * The formatting doesn't follow PocketMine-MP's code style.

Change requested corrected normally in the previous commit

File: src/pocketmine/command/defaults/TransferServerCommand.php [hunk 1]
```diff
@@ -46,13 +49,29 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
 
 		if(count($args) < 1){
 			throw new InvalidCommandSyntaxException();
-		}elseif(!($sender instanceof Player)){
+		}elseif(count($args) === 3){
+			if (!$sender->hasPermission("pocketmine.command.transferserver.other")){
```
Comment (dktapps) [inline src/pocketmine/command/defaults/TransferServerCommand.php:53]: This permission doesn't exist. It needs to be defined in `DefaultPermissions`.

File: src/pocketmine/command/defaults/TransferServerCommand.php [hunk 2]
```diff
@@ -46,13 +49,29 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
...
+				$message = $sender->getServer()->getLanguage()->translateString(TextFormat::RED . "%commands.generic.permission");
+				$sender->sendMessage($message);
+
+				return true;
+			}
+			$target = $sender->getServer()->getPlayer($args[2]);
+			if (!$target instanceof Player || !$target->isOnline()){
```
Comment (dktapps) [inline src/pocketmine/command/defaults/TransferServerCommand.php:60]: getPlayer() will never return offline players.

File: src/pocketmine/command/defaults/TransferServerCommand.php [hunk 3]
```diff
@@ -46,13 +49,29 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
...
 			$sender->sendMessage("This command must be executed as a player");
 
-			return false;
+			return true;
+		}else{
+			$target = $sender;
 		}
-
-		$sender->transfer($args[0], (int) ($args[1] ?? 19132));
+
```
Comment (dktapps) [inline src/pocketmine/command/defaults/TransferServerCommand.php:73]: Unrelated formatting change.

**Comment by dktapps** [comment]
Closing this as stale due to lack of timely review - the code in this PR is now several years obsolete and would be better rewritten once PM5 is pushed to stable.

**Pull request closed without merging**